src/swarm-tui/Swarm/TUI/View.hs

```diff
@@ -538,14 +538,12 @@ drawClockDisplay lgTPS gs = hBox . intersperse (txt " ") $ catMaybes [clockWidge
   clockWidget = maybeDrawTime (gs ^. temporal . ticks) (gs ^. temporal . paused || lgTPS < 3) gs
   pauseWidget = guard (gs ^. temporal . paused) $> txt "(PAUSED)"
 
--- | Check whether the currently focused robot (if any) has a clock
---   device equipped.
+-- | Check whether the currently focused robot (if any) has some kind
+-- of a clock device equipped.
 clockEquipped :: GameState -> Bool
 clockEquipped gs = case focusedRobot gs of
   Nothing -> False
-  Just r
-    | countByName "clock" (r ^. equippedDevices) > 0 -> True
-    | otherwise -> False
```

Comment:
This might be the only instance of `if p then True else False` in repo.

I am surprised HLint did not spot it.